### ISSUE TITLE
Default parameter not accepted for conditional type although assignable to both branches

### ISSUE THREAD

**Issue by jeysal** [microsoft/TypeScript#39521]
**TypeScript Version:**  Nightly playground 2020-07-09

**Search Terms:** conditional type not assignable, conditional branch assignable, default parameter assignable
(there are so many issues related to conditional types it's hard to tell which title may be related)

**Code**

Close to my use case:

```ts
type Options<T> = {
    t: T,
    flag?: T extends string ? boolean : false,
}

function f<Y>({ t, flag = false }: Options<Y>) { }
```

As minimal as I can get it, probably still the same cause:

```ts
type Param<T> = T extends string ? boolean : boolean;
function fn<T>(param: Param<T> = false) { }
```

**Expected behavior:**

No type error (other than unused parameters).

Note: This *could* be two separate bugs.
Example 1: The default parameter should be inferred to `false`, not `boolean`, like it is in `const x = false`, so if anything the error message should be `Type 'false' is not assignable to [...]`. It also does not seem to help to default to `[...] = (false as false)` or `[...] = (false as const)`
Example 2: Even with `[...] ? boolean : boolean` TS incorrectly does not allow to default to a `boolean` or `false`.

**Actual behavior:**

Example 1:
```
Type 'boolean' is not assignable to type 'Y extends string ? boolean : false'.(2322)
```
Example 2:
```
Type 'boolean' is not assignable to type 'Param<T>'.(2322)
```

**Playground Link:** 
Example 1:
[Playground Link](https://www.typescriptlang.org/play/?ts=4.0.0-dev.20200709&ssl=1&ssc=1&pln=6&pc=51#)
Example 2:
[Playground Link](https://www.typescriptlang.org/play/?ts=4.0.0-dev.20200709&ssl=1&ssc=1&pln=6&pc=51#code/C4TwDgpgBACghgJzgWwDwBUB8UC8V1QQAewEAdgCYDOUVwCAlmQOZQD8UARgPbcA2EOGSgAuLrwFCA3ACgAZgFcyAY2ANuwuWQyYAFGEQox8JGiy4ocuHyoQAlFADeUAL5A)

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
Fix is in progress at #30639. It's a bit of a sticky wicket.

**Comment by jeysal** [comment]
Cool, looks like https://github.com/microsoft/TypeScript/issues/26933 which that PR aims to fix (among other issues) is sufficiently similar to assume that this issue will be fixed by that as well. Closing as duplicate :+1:
